library/linux/src/embedder.cc

```diff
@@ -199,6 +199,12 @@ static void GLFWClearCanvas(GLFWwindow *window) {
   glfwMakeContextCurrent(nullptr);
 }
 
+// Resolves the address of the specified OpenGL or OpenGL ES 
+// core or extension function, if it is supported by the current context.
+static void* GLFWProcResolver(void *user_data, const char *name) {
+  return (void*) glfwGetProcAddress(name);
```

Comment:
If we do end up needing this fix on the embedding side, this needs to be `static_cast<void*>(...)`. This project follow's Google's C++ style guide, which does not use C-style casts.

Reply:
I only could get it to work with `reinterpret_cast`. But please double check what I'm doing because I'm not very familiar with C++.

Reply:
I'll need to look at this again when I have more time. Having looked at the types, this is reinterpret-casting a function pointer to a void*, which I believe violates the spec—in which either the engine API is problematic, or there's a missing step here.

Reply:
I dug into this more to see what Flutter code this was ultimately interacting with and how they handle the mismatch. It turns out that it eventually comes down to `dlsym`, so this is the issue described in the Rationale section [here](http://pubs.opengroup.org/onlinepubs/009695399/functions/dlsym.html). Pushing the requirement for the non-standards-compliant conversion up to this abstract layer, rather than keeping it closer to the `dlsym` that requires POSIX, does seem like an engine API problem.

However, since the bug is breaking a bunch of people, I'm going to go ahead and land this, and then follow up with the Flutter team separately about whether there should be more changes here at the engine level. Thanks for tracking down the issue!